### PR TITLE
added pwa sample app

### PR DESCRIPTION
**Pull request by purplecabbage**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: progressive-web-app/.github/workflows/deploy_prod.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  release:
+    types: [released]
+jobs:
+  deploy:
+    name: Deploy to Prod
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 1
+      matrix:
+        node-version: ['16']
```
Comment (shazron) [inline progressive-web-app/.github/workflows/deploy_prod.yml]: might as well use the latest?

File: progressive-web-app/.github/workflows/deploy_prod.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: npm install
+        run: npm i
+      - name: Setup CLI
+        uses: adobe/aio-cli-setup-action@1.1.0
+        with:
+          os: ${{ matrix.os }}
+          version: 9.x.x
```
Comment (shazron) [inline progressive-web-app/.github/workflows/deploy_prod.yml]: use v10?

File: progressive-web-app/.github/workflows/deploy_stage.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    branches:
+      - main
+jobs:
+  deploy:
+    name: Deploy to Stage
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 1
+      matrix:
+        node-version: ['16']
```
Comment (shazron) [inline progressive-web-app/.github/workflows/deploy_stage.yml]: use latest?

File: progressive-web-app/.github/workflows/deploy_stage.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: npm install
+        run: npm i
+      - name: Setup CLI
+        uses: adobe/aio-cli-setup-action@1.1.0
+        with:
+          os: ${{ matrix.os }}
+          version: 9.x.x
```
Comment (shazron) [inline progressive-web-app/.github/workflows/deploy_stage.yml]: use v10?

File: progressive-web-app/.github/workflows/pr_test.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
+name: AIO App CI
+
+on: [pull_request]
+jobs:
+  test:
+    name: Test PR
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        node-version: ['16']
```
Comment (shazron) [inline progressive-web-app/.github/workflows/pr_test.yml]: same

File: progressive-web-app/.github/workflows/pr_test.yml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: npm install
+        run: npm i
+      - name: Setup CLI
+        uses: adobe/aio-cli-setup-action@1.1.0
+        with:
+          os: ${{ matrix.os }}
+          version: 9.x.x
```
Comment (shazron) [inline progressive-web-app/.github/workflows/pr_test.yml]: same

File: progressive-web-app/src/dx-excshell-1/web-src/src/utils.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+ * Invokes a web action
+ *
+ * @param  {string} actionUrl
+ * @param {object} headers
+ * @param  {object} params
+ *
+ * @returns {Promise<string|object>} the response
+ *
+ */
+
```
Comment (shazron) [inline progressive-web-app/src/dx-excshell-1/web-src/src/utils.js:25]: nit: newline shouldn't be here

File: progressive-web-app/src/dx-excshell-1/web-src/src/sw.js
Comment (shazron) [inline progressive-web-app/src/dx-excshell-1/web-src/src/sw.js:1]: should this be empty?

**Review by MichaelGoberling — APPROVED** [review]

File: progressive-web-app/.github/workflows/deploy_prod.yml (hunk 1 above)
Comment (purplecabbage) [reply progressive-web-app/.github/workflows/deploy_prod.yml]: the workflows are not really required to demonstrate how this code works, so I'll just remove them all.

File: progressive-web-app/src/dx-excshell-1/web-src/src/utils.js (hunk 7 above)
Comment (purplecabbage) [reply progressive-web-app/src/dx-excshell-1/web-src/src/utils.js:25]: this is in the template, and every app created with it.

File: progressive-web-app/src/dx-excshell-1/web-src/src/sw.js
Comment (purplecabbage) [reply progressive-web-app/src/dx-excshell-1/web-src/src/sw.js:1]: added a sample service worker

**Pull request merged**